packages/ecmascript/src/Map.ts

```diff
@@ -56,26 +56,29 @@ export default class Map<K extends (string | number), V> {
 		return values;
 	}
 
-	has(key): boolean {
+	public has(key: any): boolean {
 		return this.items.hasOwnProperty(key);
 	}
 
-	get(key): any {
+	public get(key: any): any {
 		return this.items[key];
 	}
 
-	set(key, value): any {
-		let exist = this.items.hasOwnProperty(key);
+	public set(key: any, value: any): any {
+		let exist = this.has(key);
 		if (exist) {
+			this.items[key] = value;
 			this.size++;
+			return this;
 		}
-
 		this.items[key] = value;
+		this.size++;
```

Comment:
We do this regardless if it exists or not, correct?

Reply:
Yes, however there is a corner case with an existing item, hence the code sets the value, increases the size and returns.